src/app/hooks/useClickTracker/clickTypes.js

```diff
@@ -0,0 +1,42 @@
+const isLeftClick = button => button === 0;
+const isMiddleClick = button => button === 1 || button === 4; // middle click for IE is 4
+
+const isClickWithShiftOnly = event =>
+  isLeftClick(event.button) &&
+  event.shiftKey &&
+  !(event.metaKey || event.altKey || event.ctrlKey);
+
+const isCommonOpenClicked = event =>
+  isMiddleClick(event.button) || isClickWithShiftOnly(event);
+
+const isSupportedClickModifier = (event, functionKey) =>
+  (functionKey && !event.shiftKey && !event.altKey) || // functionKey
+  (event.shiftKey && functionKey && !event.altKey) || // shift + functionKey
+  (event.shiftKey && functionKey && event.altKey) || // shift + functionKey + option/alt
+  (event.shiftKey && event.altKey && !functionKey) || // shift + option/alt
+  (event.altKey && functionKey && !event.shiftKey); // option/alt + cmd
+
+const isMacOsOpenClicked = event =>
+  isLeftClick(event.button) && isSupportedClickModifier(event, event.metaKey); // cmd
+
+const isWindowsOpenClicked = event =>
+  isLeftClick(event.button) && isSupportedClickModifier(event, event.ctrlKey); // ctrl
+
+const isMacOs = () =>
+  window && window.navigator && window.navigator.platform.startsWith('Mac');
+
+export const isSafari = () =>
+  window.navigator.vendor && window.navigator.vendor.startsWith('Apple');
```

Comment:
Are we sure this function is never called on the server, where `window` will be undefined?

Reply:
@ryanmccombe This will only ever be run as part of an event handler so I'm pretty sure it won't be called on the server, but I can add optional chaining just in case 👍

Reply:
Cool :) Looks good!

Ramda is pretty good for safely handling these deep tests too: 

```javascript
pathSatisfies(startsWith('Mac'), ['navigator', 'platform'], window)
```

https://ramdajs.com/docs/#pathSatisfies
https://ramdajs.com/docs/#startsWith